src/OpenLoco/Map/BuildingTile.cpp

```diff
@@ -1,24 +1,55 @@
+#include "../CompanyManager.h"
 #include "../Game.h"
 #include "../GameCommands/GameCommands.h"
 #include "../Objects/BuildingObject.h"
 #include "../StationManager.h"
 #include "../TownManager.h"
+#include "../Ui/WindowManager.h"
 #include "../ViewportManager.h"
 #include "AnimationManager.h"
 #include "Tile.h"
 #include "TileManager.h"
 
 namespace OpenLoco::Map
 {
+    static loco_global<uint32_t, 0x01135C38> dword_1135C38;
+
+    // 0x00497DC1
     static void sub_497DC1(const Map::Pos2& loc, uint32_t population, uint32_t unk1, uint16_t rating, uint16_t unk3)
     {
-        registers regs;
-        regs.edi = population;
-        regs.esi = unk1;
-        regs.ebp = rating | (unk3 << 16);
-        regs.ax = loc.x;
-        regs.cx = loc.y;
-        call(0x00497DC1, regs);
+        auto res = TownManager::getClosestTownAndUnk(loc);
+        if (res == std::nullopt)
+        {
+            dword_1135C38 = 0xFFFFFFFF;
```

Comment:
this should be a return value from sub_497DC1. Make it `std::optional<TownId>` add a note that it is also returned via 1135c38.

Reply:
I changed it. Let me know if this is what you meant.